src/plugins/discover/public/application/main/utils/update_saved_search.ts

```diff
@@ -52,7 +52,7 @@ export function updateSavedSearch({
 
     savedSearch.searchSource
       .setField('query', state.query ?? undefined)
-      .setField('filter', [...appFilters, ...globalFilters]);
+      .setField('filter', [...globalFilters, ...appFilters]);
```

Comment:
In UI pinned filters are rendered first, then app filters. So we better pass them to Elasticsearch in the same order.

Reply:
Following up about this topic, I just wanted to add some information, because it seems the order of query / filters doesn't matter from ES side, according to https://www.elastic.co/de/blog/elasticsearch-query-execution-order 

_Q: Does the order in which I put my queries/filters in the query DSL matter?_
_A: No, because they will be automatically reordered anyway based on their respective costs and match costs._

Reply:
Thanks, @kertal! Then it's just helpful for the diffing logic.